src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -726,10 +731,10 @@ class Cluster[F[_]](
         state <- nodeState.get
         _ <- if (NodeState.canJoin(state)) F.unit else F.raiseError(new Throwable("Node attempted to double join"))
         _ <- clearServicesBeforeJoin()
-        _ <- attemptRegisterPeer(hp)
+        _ <- attemptRegisterPeer(hp) // I would consider making it a recursive process or something similar, so that we wait until it finishes the joining flow, we wouldn't have to wait with T.sleep(...) and we would know that initiall joining succeeded
         _ <- T.sleep(15.seconds)
-        _ <- LiftIO[F].liftIO(dao.downloadService.download())
-        _ <- broadcastOwnJoinedHeight()
+        _ <- LiftIO[F].liftIO(dao.downloadService.download()) // we are currently setting node to Ready as a part of download
+        _ <- broadcastOwnJoinedHeight() // I would consider if setting it to Ready along with broadcasting joining height isn't more correct, we can treat PendingDownload and ReadyForDownload as "JoiningInProgress" states
```

Comment:
something must be first in order but yes, node should be ready after all steps but it requires really good synchro between peers - probably node must be ready to receive all proposals and figure out a majority. afaik it will require more changes than only changing the order of commands

Reply:
Right. Yeah it seems more complicated than that.